core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -530,15 +540,13 @@ public void validateIndexSettings(String indexName, Settings settings) throws In
             validationErrors.add("index must have 1 or more primary shards");
         }
         if (number_of_replicas != null && number_of_replicas < 0) {
-           validationErrors.add("index must have 0 or more replica shards");
-        }
-        if (validationErrors.isEmpty() == false) {
-            throw new IndexCreationException(new Index(indexName),
-                new IllegalArgumentException(getMessage(validationErrors)));
+            validationErrors.add("index must have 0 or more replica shards");
         }
+        return validationErrors;
+
     }
 
-    private String getMessage(List<String> validationErrors) {
+    public String getMessage(List<String> validationErrors) {
```

Comment:
seems like this method doesn't belong here anymore, as it's shared between different classes. Seems it should be moved to some validation/exception util class. Actually, I am wondering if we can reuse the existing `ActionRequestValidationException` or something along those lines.